src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -68,6 +68,68 @@ void appendStyled(FlutterLogEntry entry, String text) {
       abstract void render(FlutterLogEntry entry);
     }
 
+    abstract class Column extends ColumnInfo<DefaultMutableTreeNode, FlutterLogEntry> {
+      boolean show = true;
+      private TableCellRenderer renderer;
+
+      Column(String name) {
+        super(name);
+      }
+
+      boolean isVisible() {
+        return show;
+      }
+
+      @Nullable
+      @Override
+      public FlutterLogEntry valueOf(DefaultMutableTreeNode node) {
+        if (node instanceof FlutterEventNode) {
+          return ((FlutterEventNode)node).entry;
+        }
+        return null;
+      }
+
+      @Override
+      public TableCellRenderer getCustomizedRenderer(DefaultMutableTreeNode o, TableCellRenderer renderer) {
+        if (renderer == null) {
+          renderer = createRenderer();
+        }
+        return renderer;
```

Comment:
For me, I think we can make `render` as a `final` variable then we will init it in the constructor.
```java
      @NotNull
      private final TableCellRenderer renderer;

      Column(String name) {
        super(name);
        renderer = createRenderer();
      }
```

So in this method we can simply
```java
      @Override
      public TableCellRenderer getCustomizedRenderer(DefaultMutableTreeNode o, TableCellRenderer renderer) {
        return renderer;
      }
```

Reply:
There's something clumsy here I agree.  I have some thoughts about this but let's follow-up on a bigger round of cleanup.